### ISSUE TITLE
Race condition in Tree.node()

### ISSUE THREAD

**Issue by lukaszachy** [psss/fmf#96]
Try this
```python
import fmf
import threading

def checkout():
    fmf.base.Tree.node( {
            "url": "https://github.com/psss/fmf",
            "name": "/tests/docs"
        } ).name == '/tests/docs'

threads = []
for i in range(10):
    threads.append(threading.Thread(target=checkout))

for t in threads:
    t.start()

for t in threads:
    t.join()
```

And you will see a lot of errors:
1. `Command 'git clone https://github.com/psss/fmf /home/lzachar/.cache/fmf/https:__github.com_psss_fmf' returned non-zero exit status 128.` very likely to be 'directory exists and is non-empty'
2. `Command 'git checkout -f master' returned non-zero exit status 1.`

**Comment by lukaszachy** [comment]
@psss IMO this should be fixed before v1.0

**Comment by lukaszachy** [comment]
I hit this in testsuite (running tests in parallel to speed it up) which was trying to clone same repo.
Also if TMT is able to run in parallel / on same machine for users we will hit this
